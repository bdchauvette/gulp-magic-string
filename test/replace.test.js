const { replace } = require("..")
const { fibonacci } = require("./fixtures")
const { transform } = require("./helpers")

describe("string pattern", () => {
  describe("string replacement", () => {
    it("replaces matching substrings with the provided value", async () => {
      const file = fibonacci.toVinyl()
      const transformedFile = await transform(file, replace("fibonacci", "fib"))
      expect(transformedFile.contents.toString()).toMatchSnapshot()
    })
  })

  describe("function replacement", () => {
    it("calls the function with (substring, file)", async () => {
      const file = fibonacci.toVinyl()
      const replacement = jest.fn().mockReturnValue("fib")
      await transform(file, replace("fibonacci", replacement))
      expect(replacement).toHaveBeenLastCalledWith("fibonacci", file)
    })

    it("replaces matching substrings with the return value of the provided function", async () => {
      const file = fibonacci.toVinyl()
      const replacement = (substr) => substr.toUpperCase()
      const transformedFile = await transform(
        file,
        replace("fibonacci", replacement)
      )
      expect(transformedFile.contents.toString()).toMatchSnapshot()
    })
  })
})

describe("RegExp pattern", () => {
  describe("string replacement", () => {
    it("replaces matching substrings with the provided value", async () => {
      const file = fibonacci.toVinyl()
      const transformedFile = await transform(
        file,
        replace(/fibonacci/g, "fib")
      )
      expect(transformedFile.contents.toString()).toMatchSnapshot()
    })
  })

  describe("function replacement", () => {
    it("calls the function with (substring, file)", async () => {
      const file = fibonacci.toVinyl()
      const replacement = jest.fn().mockReturnValue("fib")
      await transform(file, replace(/fibonacci/g, replacement))
      expect(replacement).toHaveBeenLastCalledWith("fibonacci", file)
    })

    it("replaces matching substrings with the return value of the provided function", async () => {
      const file = fibonacci.toVinyl()
      const replacement = (substr) => substr.toUpperCase()
      const transformedFile = await transform(
        file,
        replace(/fibonacci/g, replacement)
      )
      expect(transformedFile.contents.toString()).toMatchSnapshot()
    })
  })
})
