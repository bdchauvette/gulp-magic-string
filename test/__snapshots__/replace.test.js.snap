// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegExp pattern function replacement replaces matching substrings with the return value of the provided function 1`] = `
"export function FIBONACCI(num) {
  if (num > 1) {
    return FIBONACCI(num - 1) + FIBONACCI(num - 2)
  }

  return 1
}
"
`;

exports[`RegExp pattern string replacement replaces matching substrings with the provided value 1`] = `
"export function fib(num) {
  if (num > 1) {
    return fib(num - 1) + fib(num - 2)
  }

  return 1
}
"
`;

exports[`string pattern function replacement replaces matching substrings with the return value of the provided function 1`] = `
"export function FIBONACCI(num) {
  if (num > 1) {
    return FIBONACCI(num - 1) + FIBONACCI(num - 2)
  }

  return 1
}
"
`;

exports[`string pattern string replacement replaces matching substrings with the provided value 1`] = `
"export function fib(num) {
  if (num > 1) {
    return fib(num - 1) + fib(num - 2)
  }

  return 1
}
"
`;
